src/core/DY.Core.Bytes.fst

```diff
@@ -729,51 +729,59 @@ let has_kem_sk_usage #cusgs = mk_has_xxx_usage extract_kem_sk
 
 noeq
 type aead_crypto_predicate {|crypto_usages|} = {
-  pred: tr:trace -> key_usage:usage{AeadKey? key_usage} -> nonce:bytes -> msg:bytes -> ad:bytes -> prop;
+  pred: tr:trace -> key_usage:usage{AeadKey? key_usage} -> key:bytes{key `has_usage tr` key_usage} -> nonce:bytes -> msg:bytes -> ad:bytes -> prop;
   pred_later:
     tr1:trace -> tr2:trace ->
-    key_usage:usage{AeadKey? key_usage} -> nonce:bytes -> msg:bytes -> ad:bytes ->
+    key_usage:usage{AeadKey? key_usage} -> key:bytes{key `has_usage tr1` key_usage} -> nonce:bytes -> msg:bytes -> ad:bytes ->
     Lemma
     (requires
-      pred tr1 key_usage nonce msg ad /\
+      pred tr1 key_usage key nonce msg ad /\
+      bytes_well_formed tr1 key /\
       bytes_well_formed tr1 nonce /\
       bytes_well_formed tr1 msg /\
       bytes_well_formed tr1 ad /\
       tr1 <$ tr2
     )
-    (ensures pred tr2 key_usage nonce msg ad)
+    (ensures 
+      key `has_usage tr2` key_usage /\
+      pred tr2 key_usage key nonce msg ad
+    )
```

Comment:
```suggestion
    (ensures pred tr2 key_usage key nonce msg ad)
```
This is proved by `has_usage_later` no?

Reply:
While it does follow from `has_usage_later`, since the `has_usage` is needed to call the `pred`, and `has_usage_later` does not trigger automatically, something like this seems to be needed to allow the `pred` to be called without getting a type error.

Reply:
I have removed it, and it still works without it in the communication layer.